planner/core/planbuilder.go

```diff
@@ -4513,8 +4513,17 @@ func (b *PlanBuilder) buildDDL(ctx context.Context, node ast.DDLNode) (Plan, err
 	case *ast.RecoverTableStmt, *ast.FlashBackTableStmt:
 		// Recover table command can only be executed by administrator.
 		b.visitInfo = appendVisitInfo(b.visitInfo, mysql.SuperPriv, "", "", "", nil)
-	case *ast.LockTablesStmt, *ast.UnlockTablesStmt:
-		// TODO: add Lock Table privilege check.
+	case *ast.LockTablesStmt:
+		user := b.ctx.GetSessionVars().User
+		for _, lock := range v.TableLocks {
+			var lockAuthErr, selectAuthErr error
+			if user != nil {
+				lockAuthErr = ErrDBaccessDenied.GenWithStackByArgs(user.AuthUsername, user.AuthHostname, lock.Table.Schema.L)
```

Comment:
I think the stack is not necessary?

Reply:
`GenWithStackByArgs`  is used everywhere else. 
But I think you are right, addressed.